### PR TITLE
feat(meta): add POST /provisionwatcher V2 API

### LINKED ISSUE #2998: [V2 meta] Implement POST /provisionwatcher V2 API
# 🚀 Feature Request

### Relevant Package

 This feature request is for metadata

### Description
implement POST /provisionwatcher V2 API

### PR DESCRIPTION
**Pull request by chr1shung**
Signed-off-by: Chris Hung <chris@iotechsys.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## Issue Number:
fix #2998 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: internal/core/metadata/v2/application/provisionwatcher.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	dbClient := v2MetadataContainer.DBClientFrom(dic.Get)
+	lc := container.LoggingClientFrom(dic.Get)
+	correlationId := correlation.FromContext(ctx)
+
+	addProvisionWatcher, err := dbClient.AddProvisionWatcher(pw)
+	if err != nil {
+		return "", errors.NewCommonEdgeXWrapper(err)
+	}
+
+	lc.Debug(fmt.Sprintf(
```
Comment (cloudxxx8) [inline internal/core/metadata/v2/application/provisionwatcher.go:32]: use Debugf

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: internal/core/metadata/v2/application/provisionwatcher.go (hunk 1 above)
Comment (hahattan) [reply internal/core/metadata/v2/application/provisionwatcher.go]: updated.

File: internal/pkg/v2/infrastructure/redis/provisionwatcher.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	ts := common.MakeTimestamp()
+	// For Redis DB, the PUT or PATCH operation will removes the old object and add the modified one,
+	// so the Created is not zero value and we shouldn't set the timestamp again.
+	if pw.Created == 0 {
+		pw.Created = ts
+	}
+	// query API will sort the result based on Modified, so even newly created device service shall specify Modified as Created
+	pw.Modified = pw.Created
```
Comment (lenny-intel) [inline internal/pkg/v2/infrastructure/redis/provisionwatcher.go:58]: Should this be time.Now() instead? I.e. if created already set...

File: internal/pkg/v2/infrastructure/redis/provisionwatcher.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	// verify if provision watcher name is unique or not
+	exists, edgexErr = objectNameExists(conn, ProvisionWatcherCollectionName, pw.Name)
+	if edgexErr != nil {
+		return addedProvisionWatcher, errors.NewCommonEdgeXWrapper(edgexErr)
+	} else if exists {
+		return addedProvisionWatcher, errors.NewCommonEdgeX(errors.KindDuplicateName, fmt.Sprintf("provision watcher name %s already exists", pw.Name), edgexErr)
+	}
+
+	ts := common.MakeTimestamp()
+	// For Redis DB, the PUT or PATCH operation will removes the old object and add the modified one,
```
Comment (lenny-intel) [inline internal/pkg/v2/infrastructure/redis/provisionwatcher.go:52]: This comment indicates that this func will also be used for PUT or PATCh, but can't do to above checks. Please rework the comment.

**Review by lenny-goodell — CHANGES_REQUESTED** [review]

File: internal/pkg/v2/infrastructure/redis/provisionwatcher.go (hunk 3 above)
Comment (hahattan) [reply internal/pkg/v2/infrastructure/redis/provisionwatcher.go]: gonna remove the comment here.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by lenny-goodell — APPROVED** [review]
LGTM

**Pull request merged**